js/contexts/socketContext.tsx

```diff
@@ -0,0 +1,66 @@
+import { reload } from "../browser";
+import { join } from "../util/channel";
+import { getMetaContent } from "../util/metadata";
+import { Socket } from "phoenix";
+import {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { z } from "zod";
+
+const SocketContext = createContext<Socket | null>(null);
+
+const MetadataJoinResponse = z.object({
+  authenticated: z.boolean(),
+  server_release: z.string(),
+});
+
+export const SocketProvider = ({ children }: { children?: ReactNode }) => {
+  const [readySocket, setReadySocket] = useState<Socket | null>(null);
+
+  useEffect(() => {
+    const clientRelease = getMetaContent("release");
+    const initialSocket = new Socket("/socket", {
+      params: {
+        release: clientRelease,
+        token: getMetaContent("guardianToken"),
+      },
+    });
+    initialSocket.connect();
+
+    initialSocket.onError((err) => {
+      console.error(err);
+    });
+    const metadataChannel = initialSocket.channel("metadata");
+    // Reload our session if the auth has expired
+    metadataChannel.on("auth_expired", () => {
```

Comment:
Is this a phoenix thing or an orbit thing? Cuz the backend in `user_socket.ex` and `metadata_channel.ex` doesn't mention the string `"auth_expired"`. I think it successfully allows joining the metadata channel even when unauthenticated, and just returns  `%{authenticated: false}` (which you check below).

It might be vestigial and deleteable.

Reply:
Yep! thanks 👍 Deleted.